packages/flutter/lib/src/material/pickers/calendar_date_picker.dart

```diff
@@ -0,0 +1,1050 @@
+    @required this.previousTooltipText,
+    @required this.title,
+    @required this.onNextPressed,
+    @required this.onPreviousPressed,
+    @required this.onTitlePressed,
+  });
+
+  /// The current display of the calendar picker.
+  final DatePickerMode mode;
+
+  /// The tooltip text for the next month button.
+  final String nextTooltipText;
+
+  /// The tooltip text for the previous month button.
+  final String previousTooltipText;
+
+  /// The text that displays the current month/year being viewed.
+  final String title;
+
+  /// The callback when the next month button is pressed.
+  final VoidCallback onNextPressed;
+
+  /// The callback when the previous month button is pressed.
+  final VoidCallback onPreviousPressed;
+
+  /// The callback when the title is pressed.
+  final VoidCallback onTitlePressed;
+
+  @override
+  _DatePickerSubheaderState createState() => _DatePickerSubheaderState();
+}
+
+class _DatePickerSubheaderState extends State<_DatePickerSubheader>
+    with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(
+      value: widget.mode == DatePickerMode.year ? 1 : 0,
+      duration: const Duration(milliseconds: 200),
+      vsync: this,
+    )..addListener(() => setState(() {
+      // Animation's value has changed here.
+    }));
+  }
+
+  @override
+  void didUpdateWidget(_DatePickerSubheader oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.mode == widget.mode) {
+      return;
+    }
+
+    if (widget.mode == DatePickerMode.year) {
+      _controller.forward();
+    } else {
+      _controller.reverse();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ColorScheme colorScheme = Theme.of(context).colorScheme;
+    final TextTheme textTheme = Theme.of(context).textTheme;
+    final Color controlColor = colorScheme.onSurface.withOpacity(0.60);
+    const double height = 52.0;
+
+    return Container(
+      padding: const EdgeInsetsDirectional.only(start: 16, end: 4),
+      height: height,
+      child: Row(
+        children: <Widget>[
+          Flexible(
+            child: Semantics(
+              // TODO(darrenaustin): localize 'Select year'
+              label: 'Select year',
+              excludeSemantics: true,
+              button: true,
+              child: Container(
+                height: height,
+                child: InkWell(
+                  onTap: widget.onTitlePressed,
+                  child: Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: 8),
+                    child: Row(
+                      children: <Widget>[
+                        Flexible(
+                          child: Text(
+                            widget.title,
+                            overflow: TextOverflow.ellipsis,
+                            style: textTheme.subtitle2?.apply(
```

Comment:
Do we really need to use 'apply' here? It's semantics are more complicated than copyWith